x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/users/40_query.yml

```diff
@@ -0,0 +1,125 @@
+---
+setup:
+  - skip:
+      features: headers
+
+  - do:
+      cluster.health:
+        wait_for_status: yellow
+
+  - do:
+      security.put_role:
+        name: "manage_users_role"
+        body: >
+          {
+            "cluster": ["manage_security"]
+          }
+
+  - do:
+      security.put_user:
+        username: "users_admin"
+        body: >
+          {
+            "password" : "x-pack-test-password",
+            "roles" : [ "manage_users_role" ],
+            "full_name" : "Read User"
+          }
+
+---
+teardown:
+  - do:
+      security.delete_role:
+        name: "manage_users_role"
+        ignore: 404
+
+  - do:
+      security.delete_user:
+        username: "users_admin"
+        ignore: 404
+
+  - do:
+      security.delete_user:
+        username: "test_user_1"
+        ignore: 404
+
+  - do:
+      security.delete_user:
+        username: "test_user_2"
+        ignore: 404
+
+---
+"Test query user":
```

Comment:
Since this is already tested heavily, this is only a subset of all tests we could add here. Happy to add more if that's useful, but I think it's just more redundant code to maintain.

Reply:
++ no need to cover edge cases here